plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1787,6 +1810,28 @@ private void checkUnsupportedGeneratedColumns(MetadataEntry metadataEntry)
         }
     }
 
+    private void checkUnsupportedColumnMappingIdStructuralTypes(MetadataEntry metadataEntry)
+    {
+        ColumnMappingMode columnMappingMode = getColumnMappingMode(metadataEntry);
+        boolean hasTableStructuralTypes = getColumns(metadataEntry).stream()
+                .filter(column -> HiveUtil.isStructuralType(column.getPhysicalType())).findFirst().isPresent();
+
+        if (columnMappingMode == ColumnMappingMode.ID && hasTableStructuralTypes) {
+            throw new TrinoException(NOT_SUPPORTED, "Writing nested structures to tables with column mapping id is not supported");
```

Comment:
"nested structures" isn't common in Trino. I would rephrase as "row type".

Reply:
ok. done